tools/test-npm.sh

```diff
@@ -3,11 +3,11 @@
 set -e
 
 # always change the working directory to the project's root directory
-cd $(dirname $0)/..
+cd $(dirname ${BASH_SOURCE[0]})/..
```

Comment:
see: http://stackoverflow.com/a/9107028/1279026 (related comment linked below in the "get absolute path" part).